src/halse_se05x.c

```diff
@@ -616,6 +616,9 @@ static void halse_se05x_close(struct halse_dev *device)
 	dev->i2c_dev = NULL;
 	halgpio_close(dev->gpio_dev);
 	dev->gpio_dev = NULL;
+	if (dev->atr)
+		free(dev->atr);
+	dev->atr = NULL;
 }
```

Comment:
No need for the if. free() accepts NULL pointers (see also halse_se05x_warm_reset_dev()).

Reply:
you learn something new every day